### PR TITLE
Feature/7692 display one off cost row

### PR DESCRIPTION
**Pull request by alibombaye**
Story [AB#7692](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7692)
Task [AB#8683](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/8683)

Screenshot 2020-07-23 at 13 44 10

### CODE REVIEW

File: app/helpers/api/ordapi/getOrder.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const reducer = (dict, serviceRecipient) => (
+    {
+      ...dict,
+      [serviceRecipient.odsCode]: serviceRecipient,
+    });
+
+  return serviceRecipients.reduce(reducer, {});
+};
+
+const isOneOff = orderItem => orderItem.catalogueItemType === 'AssociatedService'
```
Comment (sam-roberts-bjss) [inline app/helpers/api/ordapi/getOrder.js:17]: Probably don't need to but might want to consider ignoring the case of the comparisons here.

File: app/pages/preview/controller.js [hunk 2]
```diff
@@ -1,25 +1,7 @@
...
-
-export const getPreviewPageContext = ({ orderId, orderData }) => {
-  const serviceRecipients = createServiceRecipientsDict(orderData.serviceRecipients);
-  const { recurringCostItems } = transformOrderItems(orderData.orderItems);
-
-  return getContext({
-    orderId, orderData, recurringCostItems, serviceRecipients,
-  });
-};
+export const getPreviewPageContext = ({
```
Comment (sam-roberts-bjss) [inline app/pages/preview/controller.js:3]: 👍  Makes this much simpler 👍

File: app/pages/preview/template.test.js [hunk 3]
```diff
@@ -218,6 +228,39 @@ describe('preview page', () => {
...
+        const recipientNameDetails = oneOffCost1Row.find('div[data-test-id="recipient-name"]');
+        const itemId = oneOffCost1Row.find('div[data-test-id="item-id"]');
+        const itemName = oneOffCost1Row.find('div[data-test-id="item-name"]');
+        const priceUnit = oneOffCost1Row.find('div[data-test-id="price-unit"]');
+        const quantity = oneOffCost1Row.find('div[data-test-id="quantity"]');
+        const itemCost = oneOffCost1Row.find('div[data-test-id="item-cost"]');
+
+        expect(oneOffCost1Row.length).toEqual(1);
+
+        expect(recipientNameDetails.length).toEqual(1);
```
Comment (sam-roberts-bjss) [inline app/pages/preview/template.test.js:245]: Is this similar to doing the OK checks that were removed? Are the length checks needed? I suppose these are more of a unit test so we aren't affected too much by them.

Comment (alibombaye) [reply app/pages/preview/template.test.js:245]: yeah you are right. They are redundant but these are pretty quick to run though. Will leave them in for now.

**Pull request merged**